sklearn/decomposition/kernel_pca.py

```diff
@@ -259,7 +262,7 @@ def inverse_transform(self, X):
         "Learning to Find Pre-Images", G BakIr et al, 2004.
         """
         if not self.fit_inverse_transform:
-            raise ValueError("Inverse transform was not fitted!")
+            raise NotFittedError("Inverse transform was not fitted.")
```

Comment:
We could introduce a new exception subclass but I think this too much. Maybe we could just improve the exception message to make it explicit that the parameter `fit_inverse_transform` should be enabled when fitting the model to be able to use this method.